### PR TITLE
run pre-commit in gha workflows

### PR DESCRIPTION
**Pull request by onerandomusername**
closes GH-154

### CODE REVIEW

**Comment by gustavwilliam** [comment]
@Akarys42 I have no idea how this works. Mind taking a quick look?

File: .github/workflows/lint-build-deploy.yml [hunk 1]
```diff
@@ -48,11 +55,16 @@ jobs:
       # Format used:
       # ::error file={filename},line={line},col={col}::{message}
       - name: Run flake8
+        # this check ensures that black always runs if the pre-commit step ran
+        # Its best to only have to fix pre-commit once than twice
+        if: always() && (steps.pre-commit.outcome == 'success' || steps.pre-commit.outcome == 'failure')
```
Comment (Akarys42) [inline .github/workflows/lint-build-deploy.yml:60]: What is that always doing here?

Comment (gustavwilliam) [reply .github/workflows/lint-build-deploy.yml:60]: Making sure that the universe hasn't ended, of course.

Comment (onerandomusername) [reply .github/workflows/lint-build-deploy.yml:60]: > What is that always doing here? 

without the always, the rest of the statement doesn't always evaluate for me. it's weird, but I think there's some internal checking for always(), so it's there...

**Review by Akarys42 — APPROVED** [review]

**Pull request merged**